source/menu/GameBrowseMenu.cpp

```diff
@@ -1689,18 +1693,22 @@ void GameBrowseMenu::UpdateCallback(void * e)
 
 void GameBrowseMenu::SetFreeSpace(float freespace, float used)
 {
-	if (strcmp(Settings.db_language, "JA") == 0)
-		usedSpaceTxt->SetText(fmt("%.2fGB %s %.2fGB %s", freespace + used, tr( "of" ), freespace, tr( "free" )));
+	if (strcmp(Settings.db_language, "JA") == 0) {
+		if (  allowUsedSpaceTxtUpdate == true )
+			usedSpaceTxt->SetText(fmt("%.2fGB %s %.2fGB %s", freespace + used, tr( "of" ), freespace, tr( "free" )));
+	}
 	else
-		usedSpaceTxt->SetText(fmt("%.2fGB %s %.2fGB %s", freespace, tr( "of" ), freespace + used, tr( "free" )));
+		if (  allowUsedSpaceTxtUpdate == true )
+			usedSpaceTxt->SetText(fmt("%.2fGB %s %.2fGB %s", freespace, tr( "of" ), freespace + used, tr( "free" )));
```

Comment:
Can you move the `if(allowUsedSpaceTxtUpdate)` check to wrap this whole block? That will eliminate the duplicate check

Reply:
You are right!

Reply:
Thinking about it, I've remembered now why I doubled the check. The error is a race condition, so the closer the condition is to the problematic statement, the better. I may be being too conservative, but I prefer to put the "if" right before the "maybe already deleted" method call. So I think I should revert the code to its previous form. Do you agree?

Reply:
Ehh at that point we're talking nanoseconds of difference in the race. My comment was a nitpick anyway, so I'm not super concerned about it.

I think the ideal way to handle the race would be with a mutex or semaphore, but I think that can come in a later fix. What you've done here will solve 99% of issues if I'm understanding correctly

Reply:
That's right. Despite the difference is really minuscule, if you don't mind, I would prefer to go back to the previous version.